indra/llwindow/llwindowwin32.cpp

```diff
@@ -3684,7 +3698,7 @@ S32 OSMessageBoxWin32(const std::string& text, const std::string& caption, U32 t
         break;
     }
 
-    int retval_win = MessageBoxW(NULL, // HWND
+    int retval_win = MessageBoxW(sWindowHandleForMessageBox, // HWND
```

Comment:
This looks a bit convoluted, and why have two variables do the same thing?
Suggestions:
LLWindowWin32 *window = static_cast<LLWindowWin32*>(mWindow);
HWND handle = window ? window->mHandle : 0;
MessageBoxW(handle ,...)

Reply:
> Suggestions:
> LLWindowWin32 _window = static_cast<LLWindowWin32_>(mWindow);

The function `OSMessageBox` is not a member of the class `LLWindow`
and doesn't have access neither to an instance of this class nor to it's field `mWindow`

Reply:
It should have access to gViewerWindow

Reply:
static_cast<LLWindowWin32>(gViewerWindow->getWindow())

Reply:
> It should have access to gViewerWindow->getWindow(), it's global

Maybe it should, but it can't :-)
The variable gViewerWindow is not accessible from llwindow library

I have already gone through all these ideas myself :-)

Reply:
You are correct, there is bunch of accessibility issues. Still not a nice solution, fills like it should have been a class with  setHWND and IsWindow(handle) checks, but this code being platform specific makes it's complicated...

Reply:
> I have already gone through all these ideas myself :-)

When I've tried several things that didn't work out, even if I realized they wouldn't before writing or testing the relevant code, and I write something that might make a subsequent maintainer raise their eyebrows, I often leave a comment of the form "This is why I'm doing it this way, instead of what you would think would be more obvious..."

Reply:
> I often leave a comment

Done